src/OpenTelemetry/Sdk.cs

```diff
@@ -185,7 +185,28 @@ public static TracerProvider CreateTracerProvider(Action<TracerProviderBuilder>
                 },
 
                 // Callback when Activity is stopped.
-                ActivityStopped = activityProcessor.OnEnd,
+                ActivityStopped = (activity) =>
+                {
+                    EnrichmentScope scope = EnrichmentScope.Current;
```

Comment:
I guess this can be put into an `EnrichmentActivityProcessor` which folks can optionally add/enable, instead of in the hot-path by default (that everyone has to pay the tax)?

Reply:
Great idea! I'll update soon.